sdks/python/apache_beam/io/filebasedsource.py

```diff
@@ -207,22 +212,36 @@ def split(self, desired_bundle_size, start_offset=None, stop_offset=None):
     if stop_offset is None:
       stop_offset = self._stop_offset
 
-    bundle_size = max(desired_bundle_size, self._min_bundle_size)
-
-    bundle_start = start_offset
-    while bundle_start < stop_offset:
-      bundle_stop = min(bundle_start + bundle_size, stop_offset)
+    if self._file_based_source.splittable:
+      bundle_size = max(desired_bundle_size, self._min_bundle_size)
+
+      bundle_start = start_offset
+      while bundle_start < stop_offset:
+        bundle_stop = min(bundle_start + bundle_size, stop_offset)
+        yield iobase.SourceBundle(
+            bundle_stop - bundle_start,
+            _SingleFileSource(
+                self._file_based_source,
+                self._file_name,
+                bundle_start,
+                bundle_stop,
+                min_bundle_size=self._min_bundle_size),
+            bundle_start,
+            bundle_stop)
+        bundle_start = bundle_stop
+    else:
       yield iobase.SourceBundle(
-          bundle_stop - bundle_start,
+          stop_offset - start_offset,
```

Comment:
Currently it will be zero. But I didn't want to prevent a future usecase where we only want to prevent a second round of splitting (and this seems more consistent).